dev/tools/lib/src/update_packages.dart

```diff
@@ -0,0 +1,1493 @@
+/// package version in cases when upgrading to the latest breaks Flutter.
+const Map<String, String> _kManuallyPinnedDependencies = <String, String>{
+  // Add pinned packages here.
+  // Dart analyzer does not catch renamed or deleted files.
+  // Therefore, we control the version of flutter_gallery_assets so that
+  // existing tests do not fail when the package has a new version.
+  'flutter_gallery_assets': '^0.2.0',
+  'mockito': '^4.1.0',  // Prevent mockito from downgrading to 4.0.0
+  'vm_service_client': '0.2.6+2', // Final version before being marked deprecated.
+  'video_player': '0.10.6', // 0.10.7 fails a gallery smoke test for toString.
+  'flutter_template_images': '1.0.1', // Must always exactly match flutter_tools template.
+};
+
+class UpdatePackagesCommand extends Command<void> {
+  UpdatePackagesCommand() {
+    argParser
+      ..addFlag(
+        'force-upgrade',
+        help: 'Attempt to update all the dependencies to their latest versions.\n'
+              'This will actually modify the pubspec.yaml files in your checkout.',
+        defaultsTo: false,
+        negatable: false,
+      )
+      ..addFlag(
+        'paths',
+        help: 'Finds paths in the dependency chain leading from package specified '
+              'in --from to package specified in --to.',
+        defaultsTo: false,
+        negatable: false,
+      )
+      ..addOption(
+        'from',
+        help: 'Used with flag --dependency-path. Specifies the package to begin '
+              'searching dependency path from.',
+      )
+      ..addOption(
+        'to',
+        help: 'Used with flag --dependency-path. Specifies the package that the '
+              'sought after dependency path leads to.',
+      )
+      ..addFlag(
+        'transitive-closure',
+        help: 'Prints the dependency graph that is the transitive closure of '
+              'packages the Flutter SDK depends on.',
+        defaultsTo: false,
+        negatable: false,
+      )
+      ..addFlag(
+        'consumer-only',
+        help: 'Only prints the dependency graph that is the transitive closure '
+              'that a consumer of the Flutter SDK will observe (When combined '
+              'with transitive-closure)',
+        defaultsTo: false,
+        negatable: false,
+      )
+      ..addFlag(
+        'verify-only',
+        help: 'verifies the package checksum without changing or updating deps',
+        defaultsTo: false,
+        negatable: false,
+      )
+      ..addFlag(
+        'offline',
+        help: 'Use cached packages instead of accessing the network',
+        defaultsTo: false,
+        negatable: false,
+      );
+  }
+
+  @override
+  final String name = 'update-packages';
+
+  @override
+  final String description = 'Update the packages inside the Flutter repo.';
+
+  @override
+  final List<String> aliases = <String>['upgrade-packages'];
+
+  @override
+  final bool hidden = true;
+
+
+  Future<void> _downloadCoverageData() async {
```

Comment:
Would you mind adding a few comments about why the package update command is downloading coverage data?

Reply:
done